packages/liveblocks-core/src/room.ts

```diff
@@ -510,7 +529,8 @@ export function isRoomEventName(value: string): value is RoomEventName {
     value === "event" ||
     value === "error" ||
     value === "connection" ||
-    value === "history"
+    value === "history" ||
+    value === "pending-storage-modifications"
```

Comment:
Btw, when I introduced the `EventSource` and event "hub" abstraction (as part of the 0.18 release), I definitely intended those to be public APIs that developers could use directly. Because I didn't want to call too much attention to it as part of 0.18, I kept the `Room.subscribe()` API intact in a backward-compatible way, but internally it's just a wrapper around those new (typed) event hub APIs.

I had in mind that the event hub might one day become the primary/stable API for subscribing to room events, and do it in a way that's somewhat easier to maintain (and to type with TypeScript, too). The various overloads of `Room.subscribe()` have historically been a complicated part of the client code base.

Now that we're extending the system with a new event, do we want to consider only adding this new event to the event hub only, and not "backport" it to the `Room.subscribe()` API, i.e. stop maintaining it and eventually at one point deprecate it?

The main rationale for the change was to avoid the various overloads of `room.subscribe()` in favor of a more direct subscription model. As a user, you can subscribe via `room.events.XXX.subscribe(...)` — in this case `room.events.pendingStorageModifications.subscribe(...)`.

Would you mind using that API directly, starting now?

Reply:
The events hub is a way better API than `subscribe` for the reasons you're mentioning and will become the public API at some points. However, I want to do it properly and not have events that are supported only using a single pattern until we explain how to migrate and why we're recommending this API and we adapt our examples. I don't want to avoid explaining to users why this event is only supported using this pattern until we're ready to support the events hub officially.